### PR TITLE
[parlview] add Australian Parliament VOD extractor

### PR DESCRIPTION
**Pull request by king-millez**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

---

This *pull request* adds an info extractor for the Australian Parliament's *video on demand* system: Parlview. The extractor can download archives of any type from Parlview, like a House of Representatives broadcast or Senate committee hearing.

### CODE REVIEW

File: yt_dlp/extractor/parlview.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            'url': url,
+            'title': self._search_regex(
+                r'<h2>([^<]+)<',
+                _html, 'title', fatal=False),
+            'formats': formats,
+            'timestamp': int(time.mktime(_timestamp.timetuple())) or int(_timestamp.timestamp()),
+            'description': self._search_regex(
+                r'<div[^>]+class="descripton"[^>]*>[^>]+<strong>[^>]+>[^>]+>([^<]+)', _html, 'description').strip() or self._search_regex(
+                    # The APH website has a typo of "descripton" instead of "description", so this is here in the event that the typo is fixed.
+                    r'<div[^>]+class="description"[^>]*>[^>]+<strong>[^>]+>[^>]+>([^<]+)', _html, 'description').strip(),
```
Comment (pukkandan) [inline yt_dlp/extractor/parlview.py:69]: too long. save to a variable first. Also, the regex can just have `descripti?on`

File: yt_dlp/extractor/parlview.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            'uploader': self._search_regex(
+                r'<td>[^>]+>Channel:[^>]+>([^<]+)', self._download_webpage(
+                    self._MEDIA_INFO_URL % vod_id, vod_id), 'channel').strip(),
```
Comment (pukkandan) [inline yt_dlp/extractor/parlview.py:72]: download the webpage before `return` (and add a note)

**Review by pukkandan — COMMENTED** [review]
Also, just a couple suggestions
1. If you find yourself prefixing with `_` for most of your local variables, you probably shouldn't be using `_` in the first place. For `_real_extract`, there is no real sense of a (local) variable being public or private, so `_` only makes it harder to read. Obviously, I wont force you to change them though
2. Try to use variable names that are similar to other extractors. Eg: `webpage` instead of `_html` and `video_id` instad of `vod_id`. This will make reading it easier for those already somewhat familiar with youtube-dl code

File: yt_dlp/extractor/parlview.py (hunk 2 above)
Comment (king-millez) [reply yt_dlp/extractor/parlview.py]: > (and add a note)

?

**Pull request merged**